apps/site/src/app/apply/sections/Components/TextInput.tsx

```diff
@@ -0,0 +1,39 @@
+import RequiredAsterisk from "./RequiredAsterisk";
+
+interface TextProps {
+	name: string;
+	labelClass: string;
+	labelText: string;
+	inputClass: string;
+	containerClass: string;
+	type: string;
+	placeholder: string;
+	isRequired: boolean;
+}
+
+export default function TextInput({
+	name,
+	labelClass,
+	labelText,
+	inputClass,
+	containerClass,
+	placeholder,
+	type,
+	isRequired,
+}: TextProps) {
+	return (
+		<div className={containerClass}>
+			<label className={labelClass} htmlFor={name}>
+				{`${labelText} `} {isRequired && <RequiredAsterisk />}
```

Comment:
Thought: given most of the fields on the application are required, having the asterisk everywhere seems a bit excessive. Although we don't have an official style guide, here's one such suggestion from Material Design
> If most fields are required, indicate optional fields by displaying the word “optional” in parentheses next to the label text ([Material Design: Text fields - Guidelines](https://m3.material.io/components/text-fields/guidelines))

Reply:
according to this, it seems like we either want to move the asterisk to the placeholder, or have a subscript of small text that labels it as required, but all of this would still include the asterisk on required attributes

do you think it would be a better idea to specify which ones are optional instead by placing (optional) in the placeholder text?

And it seems like having a red asterisk isn't necessarily a terrible design choice, as I've seen many other forms (particularly, google forms) use this convention on every field notating a required attribute

Reply:
Yes, I was suggesting the second as I quoted in the design guidelines. For Google Forms, I would say they are designed to need to accommodate custom forms that the user would construct and as such can't make as flexible design choices. We can revisit this and discuss with graphics as well.